torchrl/envs/transforms/transforms.py

```diff
@@ -4323,6 +4323,210 @@ def __repr__(self) -> str:
         )
 
 
+class Stack(Transform):
+    """Stacks tensors and tensordicts.
+
+    This transform is useful for environments that have multiple agents with
+    identical specs under different keys. The specs and tensordicts for the
+    agents can be stacked together under a shared key, in order to run MARL
+    algorithms that expect the tensors for observations, rewards, etc. to
+    contain batched data for all the agents.
```

Comment:
I would rather have a general statement about the transform first - some will use that just to stack tensors together, not in MARL settings (eg multiple images in one)

Reply:
maybe add a note saying that multiple stacks will require multiple transforms?

Reply:
Added, let me know if that's good